src/load_safe.test.js

```diff
@@ -45,20 +49,84 @@ describe('Add an existing safe', () => {
+    await clickByText('a', 'Add existing Safe', gnosisPage)
     await assertElementPresent(loadSafeForm.form, gnosisPage)
+
+    // [Step 1] Select Network
+    console.log('Shows the select network step')
+    await assertElementPresent(loadSafeForm.select_network_step, gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_step, 'Rinkeby', gnosisPage)
+
+    console.log('Switches the network')
+    // opens the network dialog
+    await clickByText('button', 'Switch Network', gnosisPage)
+    await assertElementPresent(loadSafeForm.select_network_dialog, gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'Mainnet', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'Rinkeby', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'XDai', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'EWC', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'Volta', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'Polygon', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_dialog, 'BSC', gnosisPage)
+
+    // selects the Volta network
+    await clickByText("div[role='button'] > span", 'Volta', gnosisPage)
+    await assertTextPresent(loadSafeForm.select_network_step, 'Volta', gnosisPage)
+
+    // selects the Rinkeby network again (this time clicking in the network label)
+    await clickByText('p > span', 'Volta', gnosisPage)
+    await assertElementPresent(loadSafeForm.select_network_dialog, gnosisPage)
+    await clickByText("div[role='button'] > span", 'Rinkeby', gnosisPage)
+
+    await clickElement(generalInterface.submit_btn, gnosisPage)
+
+    // [Step 2] Name and address
     console.log('Types name and address for the safe')
+    await assertElementPresent(loadSafeForm.name_and_address_safe_step, gnosisPage)
+
+    // loads a safe address with QR code
+    console.log('Load Safe address with a QR code')
+    gnosisPage.waitForFileChooser()
+    await gnosisPage.evaluate(() => {
+      return document.querySelector("img[role='button']").click()
+    })
+    await clickByText('button > span', 'Upload an image', gnosisPage)
+    await gnosisPage.waitForTimeout(2000)
+    await gnosisPage.waitForSelector("input[type='file']")
+    const button = await gnosisPage.$("input[type='file']")
+    await button.uploadFile(safeQRCodeFilePath)
+    await gnosisPage.waitForTimeout(2000)
+    await gnosisPage.keyboard.press('Escape')
+    const safeAddress = await getInnerText(loadSafeForm.safe_address_field, gnosisPage)
+    expect(safeAddress).toBe('0x57CB13cbef735FbDD65f5f2866638c546464E45F')
+    await assertElementPresent(loadSafeForm.valid_address, gnosisPage)
+
+    // Invalid Address validation
+    console.log('Invalid Safe address validation')
+    const invalidAddressError = 'Must be a valid address, ENS or Unstoppable domain'
```

Comment:
We already have this error message in /selectors/errorMsg.js if you want to use it

Reply:
done! I added this error: `invalid_Safe_Address: 'Address given is not a valid Safe address'`